app/services/post_alerter.rb

```diff
@@ -178,12 +178,19 @@ def unread_posts(user, topic)
                SELECT last_read_post_number FROM topic_users tu
                WHERE tu.user_id = ? AND tu.topic_id = ? ),0)',
                 user.id, topic.id)
-      .where('reply_to_user_id = ? OR exists(
-            SELECT 1 from topic_users tu
-            WHERE tu.user_id = ? AND
-              tu.topic_id = ? AND
-              notification_level = ?
-            )', user.id, user.id, topic.id, TopicUser.notification_levels[:watching])
+      .where('reply_to_user_id = ?
+        OR exists(SELECT 1 from topic_users tu
+                  WHERE tu.user_id = ? AND
+                    tu.topic_id = ? AND
+                    notification_level = ?)
+        OR exists(SELECT 1 from category_users cu
+                  WHERE cu.user_id = ? AND
+                    cu.category_id = ? AND
+                    notification_level = ?)',
+        user.id,
```

Comment:
You should use named parameters, because positional parameters are very unfriendly when there are more than 3.